utils/firestore.js

```diff
@@ -392,6 +392,36 @@ const retrieveParticipantsEligibleForIncentives = async (siteCode, roundType, is
     }
 }
 
+const removeDocumentFromCollection = async (connectID) => {
+    // Collections with document related to participant
+    const collectionArray = [
+        "bioSurvey_v1",
+        "clinicalBioSurvey_v1",
+        "covid19Survey_v1",
+        "menstrualSurvey_v1",
+        "module1_v1",
+        "module1_v2",
+        "module2_v1",
+        "module2_v2",
+        "module3_v1",
+        "module4_v1",
+        "ssn",
+        "biospecimen",
+    ];
+    while (collectionArray.length > 0) {
+        const collection = collectionArray.shift();
+        const data = await db
+            .collection(collection)
+            .where("Connect_ID", "==", connectID)
+            .get();
+        if (data.size !== 0) {
+            data.docs.forEach(async (dt) => {
+                await db.collection(collection).doc(dt.id).delete();
+            });
```

Comment:
This is a strange JS nuance that can lead to bugs: `forEach` does not process `async/await` properly and can lead to unexpected behavior.

(for ... of) handles async/await properly.

reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach
Quote from reference: `forEach() expects a synchronous function — it does not wait for promises. Make sure you are aware of the implications while using promises (or async functions) as forEach callbacks.`

Reply:
@jhflorey  if i use another loop whether it will fix it? Do you have any idea?

Reply:
Yes, this does the same thing and works with async/await:
```javascript
for (const dt of data.docs) {
    await db.collection(collection).doc(dt.id).delete();
}
```

Reply:
Solved